packages/vite-plugin-cloudflare/src/index.ts

```diff
@@ -72,81 +94,135 @@ export function cloudflare<T extends Record<string, WorkerOptions>>(
 						);
 					},
 				},
-				// Ensure there is an environment for each worker
-				environments: Object.fromEntries(
-					Object.entries(pluginConfig.workers ?? {}).map(
-						([name, workerOptions]) => [
-							name,
-							createCloudflareEnvironmentOptions(workerOptions, userConfig),
-						],
-					),
-				),
 			};
 		},
 		configEnvironment(name, options) {
-			options.build = {
-				...options.build,
-				// Puts all environment builds in subdirectories of the same build directory
-				outDir: path.join(options.build?.outDir ?? 'dist', name),
-			};
-		},
-		configResolved(resolvedConfig) {
-			viteConfig = resolvedConfig;
-			normalizedPluginConfig = normalizePluginConfig(
-				pluginConfig,
-				resolvedConfig,
-			);
+			if (resolvedPluginConfig.type === 'workers') {
+				options.build = {
+					...options.build,
+					// Puts all environment builds in subdirectories of the same build directory
+					// TODO: allow the user to override this
+					outDir: path.join(options.build?.outDir ?? 'dist', name),
+				};
+			}
 		},
 		async resolveId(source) {
-			const worker = normalizedPluginConfig.workers[this.environment.name];
-			if (worker) {
-				const aliased = resolveNodeAliases(source, worker.workerOptions);
-				if (aliased) {
-					if (aliased.external) {
-						return aliased.id;
-					} else {
-						return await this.resolve(aliased.id);
-					}
-				}
+			if (resolvedPluginConfig.type === 'assets-only') {
+				return;
+			}
+
+			const workerConfig = resolvedPluginConfig.workers[this.environment.name];
+
+			if (!workerConfig) {
+				return;
+			}
+
+			const aliased = resolveNodeAliases(source, workerConfig);
+
+			if (!aliased) {
+				return;
+			}
+
+			if (aliased.external) {
+				return aliased.id;
+			} else {
+				return await this.resolve(aliased.id);
 			}
 		},
 		async transform(code, id) {
-			const worker = normalizedPluginConfig.workers[this.environment.name];
-			if (worker) {
-				const rId = await this.resolve(worker.entryPath);
-				if (id === rId?.id) {
-					return injectGlobalCode(id, code, worker.workerOptions);
+			if (resolvedPluginConfig.type === 'assets-only') {
+				return;
+			}
+
+			const workerConfig = resolvedPluginConfig.workers[this.environment.name];
+
+			if (!workerConfig) {
+				return;
+			}
+
+			const resolvedId = await this.resolve(workerConfig.main);
+
+			if (id === resolvedId?.id) {
+				return injectGlobalCode(id, code, workerConfig);
+			}
+		},
+		generateBundle() {
+			let config: RawConfig | undefined;
+
+			if (resolvedPluginConfig.type === 'workers') {
+				const workerConfig =
+					resolvedPluginConfig.workers[this.environment.name];
+
+				if (!workerConfig) {
+					return;
 				}
+
+				// Currently assuming that the entrypoint of the build is `index.js`. Should we determine this from the config?
+				workerConfig.main = 'index.js';
```

Comment:
I assume that this is configurable by the user?
Or maybe by other plugins?

So I would expect we should be able to grab this from the config somewhere...

Reply:
Yeah, I had a quick look but it wasn't obvious to me where to find this. Will look into it more.